src/spec/create_extension_spec.py

```diff
@@ -24,27 +25,133 @@ def main():
+            ),
+            NWBDatasetSpec(
+                name="contact_shapes",
+                doc="shape of the contact; e.g. 'circle'",
+                dtype="text",
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="contact_shape_params",
+                doc="size of the contact; e.g. if the contact is circular and has radius of 10um, put '10'",
+                dtype="float",
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="contact_plane_axes",
+                doc="dimension of the probe",
+                dtype="float",
+                dims=[['num_contacts', 'v1, v2', 'x,y'], ['num_contacts', 'v1,v2', 'x, y, z']],
+                shape=[[None, 2, 2], [None, 2,3]],
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="contact_positions",
+                doc="dimension of the probe",
+                dtype="float",
+                dims=[['num_contacts', 'x, y'], ['num_contacts','x, y, z']],
+                shape=[[None, 2], [None, 3]],
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="contact_annotations",
+                doc="annotation of the contact",
+                dtype="text",
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="device_channel_indices",
+                doc="ID of the channel connected to the contact",
+                dtype="int",
+                neurodata_type_inc="VectorData"
+            ),
+            NWBDatasetSpec(
+                name="electrode",
+                doc="electrode ID in ElectrodeTable; should link to ElectrodeTable",
+                neurodata_type_inc="DynamicTableRegion"
+            ),
+        ],
+        neurodata_type_inc="DynamicTable",
+        neurodata_type_def="ContactTable",
+    )
+    shank = NWBGroupSpec(
+        doc="Neural probe shanks according to probeinterface specification",
         attributes=[
             NWBAttributeSpec(
-                name='trode_id',
-                doc='The tetrode ID.',
-                dtype='int32'
+                name="shank id",
+                doc="ID of the shank in the probe; must be an integer",
+                dtype="int",
+                required=True,
+            ),
+        ],
+        groups=[
+            NWBGroupSpec(
+                doc="Neural probe contacts according to probeinterface specification",
+                neurodata_type_inc="ContactTable",
+                quantity=1
             )
         ],
+        neurodata_type_inc="NWBContainer",
+        neurodata_type_def="Shank",
+    )
+    probe = NWBGroupSpec(
+        doc="Neural probe object according to probeinterface specification",
+        attributes=[
+            NWBAttributeSpec(
```

Comment:
Add `si_units`

Reply:
I added `unit` and `conversion` attributes. I think by convention NWB files use `meter` as unit and if we want e.g. microns then we multiply it by `conversion`